simple_ldap_user/simple_ldap_user.module

```diff
@@ -626,3 +628,37 @@ function simple_ldap_user_variable_get($name, $default = NULL, $force_default =
 
   return variable_get($name, $default);
 }
+
+/**
+ * Return the user account control status under Active Directory in LDAP database
+ * Check whether the user is disabled (Active Directory only).
+ * http://support.microsoft.com/kb/305144
+ *
+ * parameter: $ldap_user
+ *  */
+function simple_ldap_useraccountcontrol_isblocked ($ldap_user) {
+  // Force to update drupal user status based on $ldap_user->useraccountcontrol
+  if ($ldap_user->server->type == 'Active Directory') {
+    if (isset($ldap_user->useraccountcontrol[0]) && (int) $ldap_user->useraccountcontrol[0] & 2) {
+      return 0; // blocked
+    }
+  }
+  return 1;
+}
+
+/**
+ * Return an user's account current status
+ *
+ * parameter: $ldap_user
+ *  */
+function simple_ldap_user_status($ldap_user) {
```

Comment:
The addition and use of this function is very clear.  It has implications, however, with how other instances of `$ldap-user->exists()` have been used in this module.  Should they be updated as well?